pkg/controller/controller.go

```diff
@@ -147,9 +147,14 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 }
 
 func getSelfDeploy(kcs kubernetes.Interface, conf *config.Config, log logr.Logger) (*appsv1.Deployment, error) {
-	deploy, err := kcs.AppsV1().Deployments(conf.NS).Get(context.Background(), conf.OperatorDeployment, metav1.GetOptions{})
+	ns := conf.OperatorNS
+	if ns == "" {
+		ns = "default"
```

Comment:
why default here and kube-system default in the flag?

Reply:
I suppose if were following the same logic as not providing a namespace means default im good with that